xcube_stac/stac.py

```diff
@@ -37,14 +49,112 @@ class Stac:
+        """Get the JSON schema for instantiating a new data store.
+
+        Returns:
+            The JSON schema for the data store's parameters.
+        """
+        stac_schema = JsonObjectSchema(
+            properties=dict(**STAC_SEARCH_ITEM_PARAMETERS),
+            required=[],
+            additional_properties=False
+        )
+        return stac_schema
+
+    def get_item_collection(
+        self, **open_params
+    ) -> Tuple[ItemCollection, List[str]]:
+        """Collects all items within the given STAC catalog
+        using the supplied *open_params*.
+
+        Returns:
+            items: item collection containing all items identified by *open_params*
+            item_data_ids: data IDs corresponding to items
+        """
+        if self._searchable:
+            open_params_mod = open_params.copy()
+            if "variable_names" in open_params_mod:
+                del open_params_mod["variable_names"]
+            if "time_range" in open_params_mod:
+                open_params_mod["datetime"] = "/".join(open_params_mod["time_range"])
+                del open_params_mod["time_range"]
+            items = self.catalog.search(**open_params_mod).item_collection()
+        else:
+            items = self._get_items_nonsearchable_catalog(
+                self.catalog,
+                **open_params
+            )
+            items = ItemCollection(items)
+        item_data_ids = self.list_item_data_ids(items)
+        return items, item_data_ids
+
+    def get_item_data_id(self, item: Item) -> str:
+        """Generates the data ID of an item, which follows the structure:
+
+            `collection_id_0/../collection_id_n/item_id`
+
+        Args:
+            item: item/feature
+
+        Returns:
+            data ID of an item
+        """
+        id_parts = []
+        pystac_object = item
+        while pystac_object.STAC_OBJECT_TYPE != "Catalog":
+            id_parts.append(pystac_object.id)
+            pystac_object = pystac_object.get_parent()
+        id_parts.reverse()
+        return self._data_id_delimiter.join(id_parts)
+
+    def get_item_data_ids(self, items: Iterable[Item]) -> Iterator[str]:
+        """Generates the data ID of an item collection,
+        which follows the structure:
+
+            `collection_id_0/../collection_id_n/item_id`
+
+        Args:
+            items: item collection
+
+        Yields:
+            data ID of an item
+        """
+        for item in items:
+            yield self.get_item_data_id(item)
+
+    def list_item_data_ids(self, items: Iterable[Item]) -> List[str]:
+        """Generates a list of data IDs for a given item collection,
+        which follows the structure:
+
+            `collection_id_0/../collection_id_n/item_id`
+
+        Args:
+            items: item collection
+
+        Returns:
+            list of data IDs for a given item collection
+        """
+        return list(self.get_item_data_ids(items))
```

Comment:
Why, I though it will be similar to `list_data_ids` implemented in xcube DataStore. Here just a special case for items. In the next PR I will go down to data ID level, which includes asset IDs.

Reply:
Because `list_data_ids()` is an implementation in the base class, it returns `list(self.get_data_ids())`.

Reply:
Yes I know, but this is not implemented for `get_item_data_ids()` and `list_item_data_ids()`. 

Note that item IDs will be given by: `collection_id_0/../collection_id_n/item_id`

And data IDs will be given by: `collection_id_0/../collection_id_n/item_id/asset_id`

Reply:
`list_item_data_ids()` is used by `get_item_collection()` which gives a `pystac.item_collection.ItemCollection`. The item IDs are listed on the side.